pkg/apis/pingcap/v1alpha1/tidbcluster.go

```diff
@@ -902,8 +902,9 @@ func (tc *TidbCluster) TiKVIsAvailable() bool {
 	return true
 }
 
-func (tc *TidbCluster) AllTiKVsAreAvailable() bool {
-	if len(tc.Status.TiKV.Stores) != int(tc.Spec.TiKV.Replicas) {
+func (tc *TidbCluster) AllTiKVsAreAvailable(tolerateSingleTiKVOutage bool) bool {
+	if (!tolerateSingleTiKVOutage && len(tc.Status.TiKV.Stores) != int(tc.Spec.TiKV.Replicas)) ||
+		(tolerateSingleTiKVOutage && int(tc.Spec.TiKV.Replicas-1) != len(tc.Status.TiKV.Stores)) {
```

Comment:
```suggestion
		(tolerateSingleTiKVOutage && int(tc.Spec.TiKV.Replicas-1) > len(tc.Status.TiKV.Stores)) {
```
you should ensure that the process can continue if the corrupted tikv is recovered and `tolerateSingleTiKVOutage` is still `true`